tools/classifier/src/main.py

```diff
@@ -78,22 +72,22 @@ def run(
             bins.append("unknown")
         bins_with_quotes = [f"'{b}'" for b in bins]
 
-        tool_llm = ToolLLM(tool=self)
-        llm = tool_llm.get_llm(adapter_instance_id=llm_adapter_instance_id)
-        if not llm:
+        usage_kwargs: dict[Any, Any] = dict()
+        usage_kwargs["workflow_id"] = self.workflow_id
+        usage_kwargs["execution_id"] = self.execution_id
+
+        try:
+            llm = LLM(
+                tool=self,
+                adapter_instance_id=llm_adapter_instance_id,
+                usage_kwargs=usage_kwargs,
+            )
+        except SdkError:
             self.stream_error_and_exit("Unable to get llm instance")
-        # Setting the callback_manager to be used
-        UNCallbackManager.set_callback_manager(
-            platform_api_key=self.get_env_or_die(ToolEnv.PLATFORM_API_KEY),
-            llm=llm,
-            workflow_id=self.workflow_id,
-            execution_id=self.execution_id,
-            adapter_instance_id=llm_adapter_instance_id,
-        )
-        max_tokens = tool_llm.get_max_tokens(reserved_for_output=50 + 1000)
+
+        max_tokens = llm.get_max_tokens(reserved_for_output=50 + 1000)
```

Comment:
@gaya3-zipstack why are we doing this? Can we add a comment?

Reply:
Sorry, did not check this one. Pre-existing code. Any idea @johnyrahul  ?

Reply:
I think this was introduced to limit the text passed LLM as context. As token limit error was getting thrown. I guess it was introduced by @muhammad-ali-e  or someone from the OG team if I am not wrong.

Reply:
@muhammad-ali-e @harini-venkataraman @tahierhussain - do you guys have any context on why we do this?